### PR TITLE
Replaced field_masking_span occurrences with respective ParseField

### LINKED ISSUE #63527: [Queries] make "Span field masking" consistent with other span queries
### Request

Rename `field_masking_span` to `span_field_masking`.

### Explanations

Currently the [Span field masking query](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/query-dsl-span-field-masking-query.html) is the only [`span_*`](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/span-queries.html) to end with `_span` instead of starting with `span_`

The query builder is coherent with this naming and should be updated accordingly
https://github.com/elastic/elasticsearch/blob/4e740c2e4a544c7e99963ccd9f64686b1f0531e1/server/src/main/java/org/elasticsearch/index/query/QueryBuilders.java#L377

Lucene had the right naming:
> The span field masking query maps to Lucene’s `SpanFieldMaskingQuery`

Of course this is a Breaking Change so it should not land before 8.0.0. This is also not critical, but this could improve the naming consistence of all span queries.

### PR DESCRIPTION
**Pull request by GrayFox1**
Closes #63527.

Replaced field_masking_span occurrences with respective ParseField. This change improves maintainability, since some hard coded Strings were replaced by the ParseField that i've defined in the FieldMaskingSpanQueryBuilder Class, besides being consistent with other span queries.

I've ran the tests with "gradlew :server:test --tests org.elasticsearch.index.*" and i would appreciate if someone could give me some feedback.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

**Comment by mark-vieira** [comment]
@elasticmachine test this pleasee

File: server/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public static FieldMaskingSpanQueryBuilder fromXContent(XContentParser parser) t
                         throw new ParsingException(parser.getTokenLocation(), "[field_masking_span] query must be of type span query");
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java:110]: Can we replace the name in these error messages as well?

File: server/src/test/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilderTests.java [hunk 2]
```diff
@@ -73,13 +74,13 @@ public void testFromJson() throws IOException {
                 "}";
         Exception exception = expectThrows(ParsingException.class, () -> parseQuery(json));
         assertThat(exception.getMessage(),
-            equalTo("field_masking_span [query] as a nested span clause can't have non-default boost value [0.23]"));
+            equalTo(NAME.getPreferredName() + " [query] as a nested span clause can't have non-default boost value [0.23]"));
     }
```
Comment (romseygeek) [inline server/src/test/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilderTests.java:79]: Can we also add a test that uses the old deprecated name, and checks for a deprecation warning? There's an `assertWarnings()` helper method that will check for specific deprecation strings.

**Review by romseygeek — COMMENTED** [review]
Thanks for opening this pull request @GrayFox1! I left a couple of comments but this looks pretty close to me.

**Comment by romseygeek** [comment]
@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java (hunk 1 above)
Comment (GrayFox1) [reply server/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java:110]: Sure, i'll fix those messages in my next commit

File: server/src/test/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilderTests.java (hunk 2 above)
Comment (GrayFox1) [reply server/src/test/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilderTests.java:79]: Sure, i'll add a new test in my next commit.

**Comment by GrayFox1** [comment]
Thanks for your feedback @romseygeek! I just updated my pull request according to your comments and ran all tests with `gradlew :server:test --tests org.elasticsearch.index.*`

File: .idea/eclipseCodeFormatter.xml [hunk 3]
```diff
@@ -11,4 +11,4 @@
       </ProjectSpecificProfile>
     </option>
   </component>
-</project>
+</project>
```
Comment (romseygeek) [inline .idea/eclipseCodeFormatter.xml:14]: Nit: can you revert this?

File: server/src/test/java/org/elasticsearch/search/SearchModuleTests.java [hunk 4]
```diff
@@ -303,7 +304,7 @@ public void testRegisterRescorer() {
             "combined_fields",
             "dis_max",
             "exists",
-            "field_masking_span",
+            NAME.getPreferredName(),
```
Comment (romseygeek) [inline server/src/test/java/org/elasticsearch/search/SearchModuleTests.java:307]: Let's just use "span_field_masking" here to keep in line with the other entries in the list.

**Review by romseygeek — CHANGES_REQUESTED** [review]
Thanks @GrayFox1, this looks great.  Two more nits and then I think we're done, as long as CI is happy.

**Comment by romseygeek** [comment]
@elasticmachine test this please

**Comment by romseygeek** [comment]
@GrayFox1 you'll need to merge in the latest version of our master branch to get tests to pass (we've had a version bump internally so all the BWC checks are complaining).

File: .idea/eclipseCodeFormatter.xml (hunk 3 above)
Comment (GrayFox1) [reply .idea/eclipseCodeFormatter.xml:14]: Sure, i guess this happened when i was trying to configure the eclipseCodeFormatter plugin, but i'll revert this in my next commit.

**Comment by GrayFox1** [comment]
Thanks @romseygeek! I just updated my pull request according to your comments, ran the tests with `gradlew :server:test --tests org.elasticsearch.index*` and all of them passed. 

There is only one test failing when i ran `gradlew :server:test --tests org.elasticsearch.search*` in the SearchModuleTests. The failure is on the testRegisteredQueries when it iterates on the expected queries list and can't find "span_field_masking", although i have added this query in the list.

I tried looking into the SearchModule to check if the QuerySpec Registration wasn't using the ParseField, but everything looks ok.

**Comment by romseygeek** [comment]
@elasticmachine test this please

**Comment by romseygeek** [comment]
Aha, this looks like it's a bug in the test - it's filtering out queries that have *any* deprecated names, not queries that have been entirely deprecated.  I'll open a separate fix for that.

**Comment by romseygeek** [comment]
I opened https://github.com/elastic/elasticsearch/pull/74906 to fix the search test issue - you can merge that in here first as well if you like so that we can see how happy CI is otherwise.

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Comment by romseygeek** [comment]
@elasticmachine ok to test

**Comment by romseygeek** [comment]
Thanks @GrayFox1!

**Comment by romseygeek** [comment]
Manually backported in fc5b80abead122b42a164fd8f80478204cb0674f

**Pull request merged**